front/src/components/Chat/Chat.tsx

```diff
@@ -0,0 +1,83 @@
+import IChat from '../../types/chat/IChat';
+import IChatDetails from '../../types/chat/IChatDetails';
+
+import { useEffect, useRef, useState } from 'react';
+import { getChatById } from '../../api/chat';
+import { ScrollArea, Text } from '@mantine/core';
+import IMessage from '../../types/chat/IMessage';
+
+interface IChatProps {
+  chat: IChat | null;
+}
+
+const Chat: React.FC<IChatProps> = ({ chat }) => {
+  const [page, setPage] = useState(1);
+  const [loading, setLoading] = useState(true);
+
+  const [keepFetching, setKeepFetching] = useState(true);
+
+  const [messages, setMessages] = useState<IMessage[]>([]);
+  const [scrollPosition, onScrollPositionChange] = useState({ x: 0, y: 0 });
+
+  const viewport = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    viewport?.current?.scrollTo({
+      top: viewport.current.scrollHeight,
+      behavior: 'smooth',
+    });
+  }, [viewport]);
+
+  useEffect(() => {
+    if (!chat) {
+      return;
+    }
+
+    if (keepFetching && scrollPosition.y === 0) {
+      setPage((old) => old + 1);
+      getChatById(chat.id, page).then((res) => {
+        if (res.data?.messages.length === 0) {
+          setKeepFetching(false);
+          return;
+        }
+
+        setMessages((old) => [...(res.data?.messages.reverse() ?? []), ...old]);
```

Comment:
```suggestion
        setMessages((old) => [...res.data?.messages.reverse(), ...old]);
```
No need to check if the array is empty because the if above is already checking that, no ?